### PR TITLE
adds decal reagents

### PR DESCRIPTION
**Pull request by Delimusca**
-gives decals a reagent holder, 250u.
-decals with reagents can be scooped up with beakers, buckets etc.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
things I can think of off the top of my head:

Vomit, ash, and blood/gibs

**Comment by Iamgoofball** [comment]
:+1:

File: code/game/objects/effects/decals/cleanable.dm [hunk 1]
```diff
@@ -4,4 +4,20 @@
...
+	if(istype(W, /obj/item/weapon/reagent_containers/glass) || istype(W, /obj/item/weapon/reagent_containers/food/drinks))
+		if(src.reagents && W.reagents)
+			if(!src.reagents.total_volume)
+				user << "<span class='notice'>[src] isn't thick enough to scoop up!</span>"
+				return
+			if(W.reagents.total_volume >= W.reagents.maximum_volume)
+				user << "<span class='notice'>[W] is full!</span>"
+				return
+			else
+				user << "<span class='notice'>you scoop up [src] into [W]!</span>"
```
Comment (Iamgoofball) [inline code/game/objects/effects/decals/cleanable.dm]: capitalize "you"

**Comment by babinsquared** [comment]
can't wait to go around scooping up blood to transfuse into people

**Comment by PKPenguin321** [comment]
does this mean my idea i had 1000 years ago of cloning gibbed people by taking blood from their gibs and giving it to botany could actually happen

**Comment by pudl** [comment]
gibs, instead, should produce meat like a meatspiked monkey. splashing synthflesh on the ground and rifling through gibs to get it

**Comment by Jordie0608** [comment]
Changelog

**Comment by Delimusca** [comment]
everything but blood replaced for now, will consolidate heating in another PR or something.

File: code/game/atoms.dm [hunk 2]
```diff
@@ -353,6 +353,8 @@ var/list/blood_splatter_icons = list()
 /atom/proc/add_vomit_floor(mob/living/carbon/M as mob, var/toxvomit = 0)
 	if( istype(src, /turf/simulated) )
 		var/obj/effect/decal/cleanable/vomit/this = new /obj/effect/decal/cleanable/vomit(src)
+		if(M.reagents)
+			M.reagents.trans_to(this, M.reagents.total_volume / 10)
```
Comment (Giacom) [inline code/game/atoms.dm]: I don't think "this" is a BYOND keyword. Maybe you meant "src", which is the equivalent.

File: code/modules/food&drinks/food/snacks_egg.dm [hunk 3]
```diff
@@ -18,8 +18,9 @@
 
 /obj/item/weapon/reagent_containers/food/snacks/egg/throw_impact(atom/hit_atom)
 	..()
-	new/obj/effect/decal/cleanable/egg_smudge(src.loc)
 	reagents.reaction(hit_atom, TOUCH)
+	var/obj/effect/decal/cleanable/eggdecal = new/obj/effect/decal/cleanable/egg_smudge(src.loc)
```
Comment (Giacom) [inline code/modules/food&drinks/food/snacks_egg.dm]: I'd recommend that the transfer is made to not be 100% efficient as the decal will still be some left on the floor when scooping it up.

**Comment by Delimusca** [comment]
argh my filesystem is collapsing or something

**Pull request closed without merging**